From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Otto=20Pfl=C3=BCger?= <otto.pflueger@abscue.de>
Date: Mon, 9 Dec 2024 07:20:28 +0100
Subject: [PATCH 045/225] power: supply: sc27xx: support linking charger in
 device tree
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Instead of just searching for chargers using a fixed list, add the
option to specify the charger supply in the device tree using the
standard 'power-supplies' binding.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>
---
 drivers/power/supply/sc27xx_fuel_gauge.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/power/supply/sc27xx_fuel_gauge.c b/drivers/power/supply/sc27xx_fuel_gauge.c
index d8de4ae3b..93125ee0a 100644
--- a/drivers/power/supply/sc27xx_fuel_gauge.c
+++ b/drivers/power/supply/sc27xx_fuel_gauge.c
@@ -565,7 +565,15 @@ static int sc27xx_fgu_get_status(struct sc27xx_fgu_data *data, int *status)
 {
 	union power_supply_propval val;
 	struct power_supply *psy;
-	int i, ret = -EINVAL;
+	int i, ret;
+
+	ret = power_supply_get_property_from_supplier(data->battery,
+						      POWER_SUPPLY_PROP_STATUS,
+						      &val);
+	if (!ret) {
+		*status = val.intval;
+		return 0;
+	}
 
 	for (i = 0; i < ARRAY_SIZE(sc27xx_charger_supply_name); i++) {
 		psy = power_supply_get_by_name(sc27xx_charger_supply_name[i]);
